src/operator/tensor/matrix_op-inl.h

```diff
@@ -189,7 +202,7 @@ inline bool ReshapeShape(const nnvm::NodeAttrs& attrs,
     << "Target: " << oshape
     << "\nSource: " << dshape;
   SHAPE_ASSIGN_CHECK(*out_attrs, 0, oshape);
-  return true;
+  return !shape_is_none(dshape) && !shape_is_none(oshape);
```

Comment:
To be safe, change to `return !shape_is_none(dshape) && !shape_is_none((*out_attrs)[0]);`